### PR TITLE
[Fix #4371] Prevent `Style/MethodName` from complaining about unary o…

### LINKED ISSUE #4371: Style/MethodName skip unary operator definition
```rb
class String
  def ~@
    self * 2
  end
end

p ~'hello'
```

### PR DESCRIPTION
**Pull request by Drenmi**
This cop would complain of not using the configured method naming scheme when defining unary operators, e.g.:

```
def ~@
  # ...
end
```

This was happening because we internally check the method name against our list of known operators, but the cop was passing the trailing `@` as well, causing in a lookup miss.

This change fixes that.

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [X] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Used the same coding conventions as the rest of the project.
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] All tests are passing.
* [X] The new code doesn't generate RuboCop offenses.
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/method_name_spec.rb [hunk 1]
```diff
@@ -49,6 +49,16 @@ def +(other)
       END
     end
 
+    it 'accepts unary operator definitions' do
+      expect_no_offenses(<<-END.strip_indent)
```
Comment (bbatsov) [inline spec/rubocop/cop/style/method_name_spec.rb]: I kind of like the convention to use `RUBY` instead o `END` for such heredocs. I think that in many regards the use of END everywhere is kind of an anti-patterns. People simply don't know they an use an more descriptive name. :-)

Comment (Drenmi) [inline spec/rubocop/cop/style/method_name_spec.rb]: Definitely agree. 🙂 I tend to follow whatever the test file I'm working on is doing, but I don't mind changing this whole file to `RUBY`. 😊

Comment (bbatsov) [inline spec/rubocop/cop/style/method_name_spec.rb]: Fair enough.

Comment (backus) [inline spec/rubocop/cop/style/method_name_spec.rb]: The other benefit of using `RUBY` is that editors and github will then syntax highlight the inside of the heredoc!

Comment (bbatsov) [inline spec/rubocop/cop/style/method_name_spec.rb]: Which editors?

Btw, we can probably add a rule in the style guide about proper delim naming or something.

Comment (backus) [inline spec/rubocop/cop/style/method_name_spec.rb]: Sublime and atom by default

image

image

looks like there is an extension for vim too https://github.com/joker1007/vim-ruby-heredoc-syntax

**Pull request merged**